test/extensions/filters/http/cache/cache_filter_integration_test.cc

```diff
@@ -297,6 +297,348 @@ TEST_P(CacheIntegrationTest, GetRequestWithBodyAndTrailers) {
   }
 }
 
+TEST_P(CacheIntegrationTest, ServeHeadRequest) {
+  useAccessLog("%RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS%");
+  // Set system time to cause Envoy's cached formatted time to match time on this thread.
+  simTime().setSystemTime(std::chrono::hours(1));
+  initializeFilter(default_config);
+
+  const Http::TestRequestHeaderMapImpl request_headers = {
+      {":method", "HEAD"},
+      {":path", absl::StrCat("/", protocolTestParamsToString({GetParam(), 0}))},
+      {":scheme", "http"},
+      {":authority", "ServeHeadRequest"}};
+  const std::string response_body(42, 'a');
+  Http::TestResponseHeaderMapImpl response_headers = {
+      {":status", "200"},
+      {"date", formatter_.now(simTime())},
+      {"cache-control", "public,max-age=3600"},
+      {"content-length", std::to_string(response_body.size())}};
+
+  // Send first request, and get response from upstream.
+  {
+    IntegrationStreamDecoderPtr response_decoder =
+        codec_client_->makeHeaderOnlyRequest(request_headers);
+    waitForNextUpstreamRequest();
+    upstream_request_->encodeHeaders(response_headers, /*end_stream=*/true);
+    response_decoder->waitForEndStream();
+    EXPECT_TRUE(response_decoder->complete());
+    EXPECT_THAT(response_decoder->headers(), IsSupersetOfHeaders(response_headers));
+    EXPECT_TRUE(response_decoder->headers().get(Http::Headers::get().Age).empty());
+    EXPECT_EQ(response_decoder->body().size(), 0);
+    EXPECT_THAT(waitForAccessLog(access_log_name_), testing::HasSubstr("- via_upstream"));
+  }
+
+  // Advance time, to verify the original date header is preserved.
+  simTime().advanceTimeWait(Seconds(10));
+
+  // Send second request, and get response from upstream, since the head requests are not stored in
+  // cache.
+  {
+    IntegrationStreamDecoderPtr response_decoder =
+        codec_client_->makeHeaderOnlyRequest(request_headers);
+    waitForNextUpstreamRequest();
+    upstream_request_->encodeHeaders(response_headers, /*end_stream=*/true);
+    response_decoder->waitForEndStream();
+    EXPECT_TRUE(response_decoder->complete());
+    EXPECT_THAT(response_decoder->headers(), IsSupersetOfHeaders(response_headers));
+    EXPECT_EQ(response_decoder->body().size(), 0);
+    EXPECT_TRUE(response_decoder->headers().get(Http::Headers::get().Age).empty());
+    // Advance time to force a log flush.
+    simTime().advanceTimeWait(Seconds(1));
+    EXPECT_THAT(waitForAccessLog(access_log_name_), testing::HasSubstr("- via_upstream"));
+  }
+}
+
+TEST_P(CacheIntegrationTest, ServeHeadFromCacheAfterGetRequest) {
+  useAccessLog("%RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS%");
+  simTime().setSystemTime(std::chrono::hours(1));
```

Comment:
nit: perhaps this line needs the same comment as in the cases above for consistency:
```c++
// Set system time to cause Envoy's cached formatted time to match time on this thread.
```